### PR TITLE
Allow user provided domain

### PR DESCRIPTION
**Pull request by chiayi**

### CODE REVIEW

File: jupyter-on-gke/iap_module/variables.tf [hunk 1]
```diff
@@ -49,4 +49,16 @@ variable "client" {
     secret = string
   })
   description = "oauth client"
+}
+
+variable "url_domain_addr" {
```
Comment (imreddy13) [inline jupyter-on-gke/iap_module/variables.tf:54]: Probably also needs a readme update.
I think we need a separate PR that either documents all the vars/options in the readme or in a different .md file referenced in the readme.

File: jupyter-on-gke/iap_module/variables.tf [hunk 2]
```diff
@@ -49,4 +49,16 @@ variable "client" {
...
   description = "oauth client"
+}
+
+variable "url_domain_addr" { 
+  type = string
+  description = "Domain provided by the user. If it's empty, we will create one for you."
+  default = ""
+}
+
+variable "url_domain_name" {
```
Comment (imreddy13) [inline jupyter-on-gke/iap_module/variables.tf:60]: What is name vs address? Where is name used?

File: jupyter-on-gke/iap_module/iap.tf [hunk 3]
```diff
@@ -44,19 +45,19 @@ resource "kubectl_manifest" "backend_config" {
...
+  depends_on = [ kubernetes_secret.my-secret ]
 }
 
 # Ingress for IAP
 resource "kubectl_manifest" "static_ingress" {
   override_namespace = var.namespace
 
   yaml_body          = templatefile("${path.module}/deployments/static-ingress.yaml", {
-    static_addr_name = "${google_compute_global_address.default.name}"
+    static_addr_name = var.url_domain_addr != "" ? var.url_domain_name : "${google_compute_global_address.default[0].name}"
```
Comment (imreddy13) [inline jupyter-on-gke/iap_module/iap.tf:58]: should this be url_domain_name?

File: jupyter-on-gke/iap_module/variables.tf (hunk 1 above)
Comment (chiayi) [reply jupyter-on-gke/iap_module/variables.tf:54]: Yeah definitely, I have a task for myself to add a definitions/"meta" file.

File: jupyter-on-gke/iap_module/variables.tf (hunk 2 above)
Comment (chiayi) [reply jupyter-on-gke/iap_module/variables.tf:60]: So the name is used by the ingress annotations. It's used to specify the global ip address.

File: jupyter-on-gke/iap_module/iap.tf (hunk 3 above)
Comment (chiayi) [reply jupyter-on-gke/iap_module/iap.tf:58]: So we don't want it to use the domain name if we are reserving an IP for them. I specified in the variable description that the domain name will only be used if they provide a domain url.

File: jupyter-on-gke/iap_module/variables.tf (hunk 1 above)
Comment (chiayi) [reply jupyter-on-gke/iap_module/variables.tf:54]: I added a note to the readme as well.

**Review by imreddy13 — APPROVED** [review]

**Pull request merged**